### PR TITLE
Change to keyIsEmpty function.  It will now comply with optional variables.

### PR DESCRIPTION
**Pull request by ryancoyne**

### CODE REVIEW

File: Sources/StORM/StORM.swift [hunk 1]
```diff
@@ -103,19 +103,37 @@ open class StORM {
...
+        case is Int.Type, is Int?.Type:
+            return (val as! Int == 0)
+        case is String.Type, is String?.Type:
+            return (val as! String).isEmpty
+        default:
+            print("[StORM] WARNING: Unexpected type for PRIMARY KEY in function: \(#function). TYPE: \(type)")
+            return false
+        }
+        
+//		if val is Int {
```
Comment (iamcam) [inline Sources/StORM/StORM.swift]: I suggest removing unused code unless it will be needed in the near future.

**Comment by ryancoyne** [comment]
Done!

File: Tests/PerfectCRUDTests/StORMTests.swift [hunk 2]
```diff
@@ -10,6 +10,113 @@ class StORMTests: XCTestCase {
 	override func setUp() {
 		super.setUp()
 	}
+    
+    func testKeyTypeIntegerNil() {
+        
+        let val : Int?=nil
```
Comment (iamcam) [inline Tests/PerfectCRUDTests/StORMTests.swift]: Minor formatting for consistency
`let val : Int? = nil`

File: Tests/PerfectCRUDTests/StORMTests.swift [hunk 3]
```diff
@@ -10,6 +10,113 @@ class StORMTests: XCTestCase {
...
+            XCTFail("Type should either be a String or Integer.")
+            print("[StORM] WARNING: Unexpected type for PRIMARY KEY in function: \(#function). TYPE: \(type)")
+            //     return false
+        }
+        
+    }
+    
+    func testKeyTypeStringNil() {
+        
+        let val : String?=nil
```
Comment (iamcam) [inline Tests/PerfectCRUDTests/StORMTests.swift]: Minor formatting for consistency
`let val : String? = nil`

**Review by iamcam — COMMENTED** [review]
Looks good. Only a couple very minor formatting fixes. After that, 👍

**Review by iamcam — APPROVED** [review]

**Pull request merged**